test/controllers/OtherReasonControllerSpec.scala

```diff
@@ -523,5 +528,33 @@ class OtherReasonControllerSpec extends SpecBase {
         }
       }
     }
+
+    "onSubmitForUploadComplete" should {
+      "the user is authorised" must {
+        "return 303 (SEE_OTHER) adding the key to the session when the body is correct " +
+          "- routing to non-JS evidence upload page when in Normal Mode" in new Setup(AuthTestModels.successfulAuthResult) {
+          val result: Future[Result] = controller.onSubmitForUploadComplete(NormalMode)(fakeRequestConverter(fakeRequestWithCorrectKeys.withJsonBody(
+            Json.parse(
+              """
+                |{
+                | "value": "no"
+                |}
+                |""".stripMargin))))
+          status(result) shouldBe SEE_OTHER
+          await(result).session.get(SessionKeys.nextFileUpload).get shouldBe "no"
+        }
+
+        "return 400 (BAD_REQUEST) when the user does not enter an option" in new Setup(AuthTestModels.successfulAuthResult) {
+          val result: Future[Result] = controller.onSubmitForWhyReturnSubmittedLate(NormalMode)(fakeRequestConverter(fakeRequestWithCorrectKeys.withJsonBody(
+            Json.parse(
+              """
+                |{
+                | "value": ""
+                |}
+                |""".stripMargin))))
+          status(result) shouldBe BAD_REQUEST
+        }
+      }
```

Comment:
Could you add tests for user not being authorised?
```
      "the user is unauthorised" when {
        "return 403 (FORBIDDEN) when user has no enrolments" in new Setup(AuthTestModels.failedAuthResultNoEnrolments) {
          val result: Future[Result] = controller.onSubmitForWhyReturnSubmittedLate(NormalMode)(fakeRequest)
          status(result) shouldBe FORBIDDEN
        }
        "return 303 (SEE_OTHER) when user can not be authorised" in new Setup(AuthTestModels.failedAuthResultUnauthorised) {
          val result: Future[Result] = controller.onSubmitForWhyReturnSubmittedLate(NormalMode)(fakeRequest)
          status(result) shouldBe SEE_OTHER
        }
      }
```

Reply:
Added them.